test/e2e/static_pods.go

```diff
@@ -0,0 +1,100 @@
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+func createStaticPod(f *framework.Framework, nodeName string, podYaml string) {
+	//create file
+	var podFile = "static-pod.yaml"
+	if err := ioutil.WriteFile(podFile, []byte(podYaml), 0644); err != nil {
+		framework.Failf("Unable to write static-pod.yaml  to disk: %v", err)
+	}
+	defer func() {
+		if err := os.Remove(podFile); err != nil {
+			framework.Logf("Unable to remove the static-pod.yaml from disk: %v", err)
+		}
+	}()
+	var dst = fmt.Sprintf("%s:/etc/kubernetes/manifests/%s", nodeName, podFile)
+	cmd := []string{"docker", "cp", podFile, dst}
+	framework.Logf("Running command %v", cmd)
+	_, err := runCommand(cmd...)
+	if err != nil {
+		framework.Failf("failed to copy pod file to node %s", nodeName)
+	}
+
+}
+
+func removeStaticPodFile(nodeName string, podFile string) {
+	cmd := []string{"docker", "exec", nodeName, "bash", "-c", "rm /etc/kubernetes/manifests/static-pod.yaml"}
+	framework.Logf("Running command %v", cmd)
+	_, err := runCommand(cmd...)
+	if err != nil {
+		framework.Failf("failed to remove pod file from node %s", nodeName)
+	}
+
+}
+
+//This test does the following
+//Applies a staticPod.yaml file to a nodes /etc/kubernetes/manifest dir
+//Expects the static pod to succeed
+var _ = ginkgo.Describe("Creating a static pod on a node", func() {
+
+	const (
+		podName string = "static-pod"
+		podFile string = "static-pod.yaml"
+	)
+
+	f := wrappedTestFramework("staticpods")
+
+	var cs clientset.Interface
+
+	ginkgo.BeforeEach(func() {
+		cs = f.ClientSet
+	})
+
+	ginkgo.It("Should successfully create a static pod", func() {
+		nodes, err := e2enode.GetBoundedReadySchedulableNodes(cs, 3)
+		framework.ExpectNoError(err)
+		if len(nodes.Items) < 1 {
+			framework.Failf("Test requires 1 Ready node, but there are none")
+		}
+		nodeName := nodes.Items[0].Name
+		podName := fmt.Sprintf("static-pod-%s", nodeName)
+		contName := "web"
+
+		ginkgo.By("copying a pod.yaml file into the /etc/kubernetes/manifests dir of a node")
+		framework.Logf("creating %s on node %s", podName, nodeName)
+		var staticPodYaml = fmt.Sprintf(`apiVersion: v1
+kind: Pod
+metadata:
+  name: static-pod
+  namespace: %s
+spec: 
+  containers: 
+    - name: web 
+      image: centos:8 
+      command: ["/bin/bash", "-c", "trap : TERM INT; sleep infinity & wait"]
+`, f.Namespace.Name)
+		createStaticPod(f, nodeName, staticPodYaml)
+		defer removeStaticPodFile(nodeName, podFile)
+		err = e2epod.WaitForPodSuccessInNamespace(f.ClientSet, podName, f.Namespace.Name)
+		if err != nil {
+			logs, logErr := e2epod.GetPodLogs(f.ClientSet, f.Namespace.Name, podName, contName)
+			if logErr != nil {
+				framework.Logf("Warning: Failed to get logs from pod %q: %v", podName, logErr)
+			} else {
+				framework.Logf("pod %s/%s logs:\n%s", f.Namespace.Name, podName, logs)
+			}
```

Comment:
@bpickard22 I had an idea: can you make the removal of the pod part of the test itself, instead of the cleanup? That way we will test/verify that the static pood removal works as expected (CNI DEL codepath).